### ISSUE TITLE
Block and transaction properties

### ISSUE THREAD

**Issue by clearloop** [zink-lang/zink#289]
introduce interfaces for https://docs.soliditylang.org/en/latest/units-and-global-variables.html#block-and-transaction-properties

for example, [Address::caller()](https://github.com/zink-lang/zink/blob/main/zink/src/primitives/address.rs#L26), just need to map these opcodes to functions

**Comment by poltao** [comment]
Hi, I'd like to try this. If I'm not mistaken, almost all general-use utility functions should be achievable through [EVM Codes](https://www.evm.codes/?fork=shanghai).

**Comment by clearloop** [comment]
> Hi, I'd like to try this. If I'm not mistaken, almost all general-use utility functions should be achievable through [EVM Codes](https://www.evm.codes/?fork=shanghai).

yes, exactly! just need to map these opcodes! btw need tests with REVM to confirm everything works!
